int32 id
bool open
